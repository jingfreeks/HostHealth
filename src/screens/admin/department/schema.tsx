import * as yup from 'yup';
import {message} from '@/config/constant'

export const Schema = yup
  .object({
    name: yup.string().required(message[100005]),
  })
  .required();
