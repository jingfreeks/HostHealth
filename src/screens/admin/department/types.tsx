import { RouteProp } from '@react-navigation/native';
import type {AppNavigationProps} from '@/navigation/types';
type StateFormScreenRouteProp = RouteProp<AppNavigationProps, 'DepartmentForm'>;
export type RoutesProps = {
  route: StateFormScreenRouteProp
};

export type StateFormProps = {
  name:string;
  _id:string;
};
