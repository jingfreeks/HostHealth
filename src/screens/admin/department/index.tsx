export {default as Department} from './department'
export {default as DepartmentForm} from './form'