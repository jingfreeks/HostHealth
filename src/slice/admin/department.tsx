import {createEntityAdapter} from '@reduxjs/toolkit';
import {apiSlice} from '@/config/apiSlice';

type departmentTypes = {_id: string; name: string};

export const departmentAdapter = createEntityAdapter<departmentTypes>({
  selectId: city => city._id,
  sortComparer: (a, b) => a.name.localeCompare(b.name),
});
export const initialState = departmentAdapter.getInitialState();
export const departmentApiSlice = apiSlice.injectEndpoints({
  endpoints: builder => ({
    getDept: builder.query({
      query: () => '/dept',
      transformResponse: responseData => {
        return departmentAdapter.setAll(initialState, responseData);
      },
      providesTags: (result: any, error, arg): any =>
        result
          ? [
              {type: 'Department', id: 'LIST'},
              ...result.ids.map((id: string | number) => ({
                type: 'Department',
                id,
              })),
            ]
          : ['Department'],
    }),
    addDept: builder.mutation({
      query: credentials => ({
        url: '/dept',
        method: 'POST',
        body: {...credentials},
      }),
      invalidatesTags: ['Department'] as string[] & undefined,
    }),
    updateDept: builder.mutation({
      query: credentials => ({
        url: '/dept',
        method: 'PATCH',
        body: {...credentials},
      }),
      invalidatesTags: ['Department'] as string[] & never & undefined,
    }),
    deleteDept: builder.mutation({
      query: credentials => ({
        url: '/dept',
        method: 'DELETE',
        body: {...credentials},
      }),
      invalidatesTags: ['Department'] as string[] & undefined,
    }),
  }),
  overrideExisting: true,
});

export const {
  useGetDeptQuery,
  useAddDeptMutation,
  useUpdateDeptMutation,
  useDeleteDeptMutation,
} = departmentApiSlice;
