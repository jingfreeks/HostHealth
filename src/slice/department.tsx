import {createEntityAdapter} from '@reduxjs/toolkit';
import {apiSlice} from '@/config/apiSlice';

type cityTypes = {_id: string; name: string};

export const cityAdapter = createEntityAdapter<cityTypes>({
  selectId: city => city._id,
  sortComparer: (a, b) => a.name.localeCompare(b.name),
});
export const initialState = cityAdapter.getInitialState();
export const departmentApiSlice = apiSlice.injectEndpoints({
  endpoints: builder => ({
    getDept: builder.query({
      query: () => '/dept',
      transformResponse: responseData => {
        return cityAdapter.setAll(initialState, responseData);
      },
      providesTags: (result: any, error, arg): any =>
        result
          ? [
              {type: 'Department', id: 'LIST'},
              ...result.ids.map((id: string | number) => ({
                type: 'Department',
                id,
              })),
            ]
          : ['Department'],
    }),
    addDept: builder.mutation({
      query: credentials => ({
        url: '/dept',
        method: 'POST',
        body: {...credentials},
      }),
      invalidatesTags: ['Department'] as any,
    }),
    updateDept: builder.mutation({
      query: credentials => ({
        url: '/dept',
        method: 'PATCH',
        body: {...credentials},
      }),
      invalidatesTags: ['Department'] as string[] & never & undefined,
    }),
    deleteDept: builder.mutation({
      query: credentials => ({
        url: '/dept',
        method: 'DELETE',
        body: {...credentials},
      }),
      invalidatesTags: ['Department'] as any,
    }),
  }),
  overrideExisting: true,
});

export const {
  useGetDeptQuery,
  useAddDeptMutation,
  useUpdateDeptMutation,
  useDeleteDeptMutation,
} = departmentApiSlice;
