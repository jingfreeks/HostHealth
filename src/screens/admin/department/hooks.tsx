import React from 'react';
import { Alert } from 'react-native';
import {StackNavigationProp} from '@react-navigation/stack';
import type {RootNavigationProps} from '@/navigation/types';
import {useNavigation} from '@react-navigation/native';
import {useGetDeptQuery, useDeleteDeptMutation} from '@/slice';

export const useDepartmentHooks = () => {
  const navigation = useNavigation<StackNavigationProp<RootNavigationProps>>();

  const {
    data: dept,
    isLoading,
    isSuccess,
    isError,
    error,
  } = useGetDeptQuery<{
    refetch: () => void;
    data: any;
    isLoading: boolean;
    isSuccess: string;
    isError: boolean;
    error: any;
  }>('getDept');

  const [deleteDept, {isLoading: deleteLoading}] = useDeleteDeptMutation();
  const handleDeleteDept = async (id: string) => {
    try {
      await deleteDept({
        id,
      }).unwrap();
    
    } catch (error) {
      Alert.alert(error?.data?.message);
    }
  };
  return {
    navigation,
    dept,
    isLoading,
    isSuccess,
    isError,
    error,
    handleDeleteDept,
    deleteLoading,
  };
};
