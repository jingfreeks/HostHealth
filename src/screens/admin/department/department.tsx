import React, {useCallback} from 'react';
import {FlatList, ListRenderItem, ListRenderItemInfo} from 'react-native';
import {FAB} from 'react-native-elements';
import {ContainerStyled, StateEmptyContainerStyled} from './styles';
import {List} from './component';
import {HomeEmptyCard} from '@/component';
import {PcitiesEmptyIcon} from '@/assets';
import {useDepartmentHooks} from './hooks';
import {testingProps} from '@/utils/testframework';
import {message} from '@/config/constant';

const Department = () => {
  let content;
  const {navigation, dept, isLoading, isSuccess, error, isError} =
    useDepartmentHooks();
  const renderItem: ListRenderItem<any> = useCallback(
    ({item}: ListRenderItemInfo<any>) => {
      return <List deptId={item} />;
    },
    [],
  );

  if (isLoading || isError) {
    let messages;
    if (error?.status === 403) {
      messages = message[100001];
    } else {
      messages = message[100004];
    }
    content = (
      <StateEmptyContainerStyled>
        <HomeEmptyCard imgsource={PcitiesEmptyIcon} message={messages} />
      </StateEmptyContainerStyled>
    );
  } else if (isSuccess) {
    content = (
      <ContainerStyled>
        <FlatList
          data={dept.ids}
          extraData={dept.ids}
          renderItem={renderItem}
          keyExtractor={(item: any, index) => index.toString()}
        />
        <FAB
          {...testingProps('DepartmentCreateButtonTestId')}
          title="Create"
          placement="right"
          size="large"
          onPress={() => navigation.navigate('DepartmentForm')}
        />
      </ContainerStyled>
    );
  }
  return content;
};

export default Department;
