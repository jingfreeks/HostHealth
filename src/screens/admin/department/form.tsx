import React from 'react';
import {
  FormHeaderContainerStyled,
  FormTextInputContainerStyled,
  FormContainerStyled,
  FormHeaderTextStyled,
} from './styles';
import {FormTextController, Bbutton} from '@/component';
import {useDepartmentHooks} from './hooks';
import {colors} from '@/utils/themes';
import {useForm, FormProvider, SubmitHandler} from 'react-hook-form';
import {yupResolver} from '@hookform/resolvers/yup';
import {useAddDeptMutation, useUpdateDeptMutation} from '@/slice';
import {Schema} from './schema';
import * as yup from 'yup';
import type {RoutesProps} from './types';

const Form = (props: RoutesProps) => {
  const {navigation} = useDepartmentHooks();
  const {route} = props;
  const {name, _id} = route?.params || {};
  type FormData = yup.InferType<typeof Schema>;
  const formMethod = useForm<FormData>({
    defaultValues: {
      name: name || '',
    },
    resolver: yupResolver(Schema),
  });

  const [addDept, {isLoading: addDeptLoading}] = useAddDeptMutation();
  const [updateDept, {isLoading: updateStateLoading}] = useUpdateDeptMutation();
  const onSubmit: SubmitHandler<FormData> = async data => {
    try {
      let response: any;
      if (_id) {
        //update
        response = await updateDept({
          name: data?.name,
          id: _id,
        }).unwrap();
      } else {
        //insert
        response = await addDept({
          name: data?.name,
        });
      }
      if (response?.error) {
        alert(response?.error?.data?.message);
      } else {
        navigation.goBack();
      }
    } catch (error) {
      console.log('error', error);
    }
  };
  return (
    <FormContainerStyled>
      <FormProvider {...formMethod}>
        <FormHeaderContainerStyled>
          <FormHeaderTextStyled TextMode="Htitlenormal">
            Department Form Information
          </FormHeaderTextStyled>
        </FormHeaderContainerStyled>
        <FormTextInputContainerStyled>
          <FormTextController
            Label="Name"
            name="name"
            placeholder="Name"
            rules={{
              required: true,
            }}
          />
        </FormTextInputContainerStyled>
        <Bbutton
          bcolor={colors.lightergreen}
          border={10}
          loaders={addDeptLoading || updateStateLoading}
          title="Save"
          onPress={formMethod.handleSubmit(onSubmit)}
        />
      </FormProvider>
    </FormContainerStyled>
  );
};

export default Form;
