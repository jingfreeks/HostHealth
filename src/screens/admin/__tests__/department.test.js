import React from 'react'
import {renderWithProviders} from '@/utils/testframeworknew';
import {Department} from '../department';

describe('Department admin screen', () => {
  it('Should work as expected to get snapshot', () => {
    const all = renderWithProviders(<Department />);
    expect(all.toJSON()).toMatchSnapshot();
  });
});
