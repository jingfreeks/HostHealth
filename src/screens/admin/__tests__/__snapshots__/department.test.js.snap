// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Department admin screen Should work as expected to get snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <RCTSafeAreaView
    style={
      [
        {},
      ]
    }
  >
    <View
      alignType="center"
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "center",
            "flexDirection": "column",
          },
        ]
      }
    >
      <Image
        resizeMode="contain"
        size={150}
        source={
          {
            "testUri": "../../../src/assets/pcitiesempty.png",
          }
        }
        style={
          [
            {
              "height": 294.2647058823529,
              "width": 294.2647058823529,
            },
          ]
        }
      />
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "Avenir",
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        No Department matches your preferred locations
      </Text>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`Department admin screen Should work as expected to get snapshot 2`] = `
<View
  style={
    [
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "marginLeft": 39.23529411764706,
        "marginRight": 39.23529411764706,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "marginBottom": 39.23529411764706,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#0e3f6c",
            "fontFamily": "Avenir",
            "fontSize": 35.311764705882354,
            "fontWeight": "500",
          },
          {
            "fontSize": 30,
            "fontWeight": "900",
          },
        ]
      }
    >
      Department Form Information
    </Text>
  </View>
  <View
    style={
      [
        {
          "marginBottom": 9.808823529411764,
          "marginTop": 39.23529411764706,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "marginLeft": 5,
          "marginRight": 5,
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 5,
        },
      ]
    }
  >
    <View
      border={10}
      borderc="#99E1E1"
      borderw={2}
      style={
        [
          {
            "borderColor": "#99E1E1",
            "borderRadius": 10,
            "borderWidth": 2,
          },
        ]
      }
    >
      <View
        accessibilityLabel="AtomsButtonTestId"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#d6f3f3",
            "borderRadius": 10,
            "opacity": 1,
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
          }
        }
        testID="AtomsButtonTestId"
      >
        <Text
          style={
            [
              {
                "color": "#03606A",
                "fontFamily": "Avenir",
                "fontSize": 15,
                "fontWeight": "500",
              },
            ]
          }
        >
          Save
        </Text>
      </View>
    </View>
  </View>
</View>
`;
